src/cljs/ataru/virkailija/editor/handlers.cljs

```diff
@@ -165,6 +166,84 @@
+                        (and (re-matches numeric-matcher clean)
+                             (<= (number-of-decimals clean) decimals)))))
+
+(defn- validate-min-max [min max decimals]
+  (let [min (format-range min)
+        max (format-range max)
+        min-valid?  (valid-range? min decimals)
+        max-valid?  (valid-range? max decimals)
+        empty-range?   (and (not (empty? min))
+                            (not (empty? max))
+                            (and min-valid? max-valid?)
+                            (gte min max))]
+    (if empty-range?
+      [false false]
+      [min-valid? max-valid?])))
+
+(reg-event-db
+  :editor/set-decimals-value
+  (fn [db [_ id value & path]]
+    (let [path             (concat path [:params])
+          min-value        (get-in db (current-form-content-path db [path :min-value]))
+          max-value        (get-in db (current-form-content-path db [path :max-value]))
+          [min? max?]      (validate-min-max min-value max-value value)]
+      (cond-> (-> db
+                  (assoc-in (current-form-content-path db [path :decimals]) value)
+                  (update-modified-by path)
+                  (assoc-in [:editor :ui id :min-value :invalid?] (not min?))
+                  (assoc-in [:editor :ui id :max-value :invalid?] (not max?)))
+
+
+              (not min?)
+              (update-in (current-form-content-path db path) (fn [params]
+                                                               (dissoc params :min-value)))
+
+              (not max?)
+              (update-in (current-form-content-path db path) (fn [params]
+                                                               (dissoc params :max-value)))))))
+
+(reg-event-db
+  :editor/set-range-value
+  (fn [db [_ id range value & path]]
+    (let [path           (concat path [:params])
+          min-range?     (= :min-value range)
+          decimals       @(subscribe [:editor/get-component-value path :decimals])
+          clean-value    (format-range value)
+          valid?         (valid-range? clean-value decimals)
+          opposing-range (if min-range?
+                           :max-value
+                           :min-value)
+          opposing-value (get-in db (current-form-content-path db [path opposing-range]) "")
+          [min? max?] (if min-range?
+                        (validate-min-max value opposing-value decimals)
+                        (validate-min-max opposing-value value decimals))]
+      (cond-> (-> db
+                  (update-modified-by path)
+                  (assoc-in [:editor :ui id range :value] value)
+                  (assoc-in [:editor :ui id :min-value :invalid?] (not min?))
+                  (assoc-in [:editor :ui id :max-value :invalid?] (not max?)))
+
+              (empty? clean-value)
+              (update-in (current-form-content-path db path) (fn [params]
+                                                                 (dissoc params range)))
+
```

Comment:
Täs pitäis kanssa poistaa `:min-value` ja `:max-value` paramsseista jos ne on epävalideja.